homeassistant/data_entry_flow.py

```diff
@@ -84,7 +84,8 @@ class FlowResultType(StrEnum):
 }
 
 
-_FlowResultT = TypeVar("_FlowResultT", bound="FlowResult")
+_FlowResultT = TypeVar("_FlowResultT", bound="BaseFlowResult")
```

Comment:
```suggestion
_FlowResultT = TypeVar("_FlowResultT", bound="BaseFlowResult[Any]")
```
`BaseFlowResult` is generic. Technically the `bound` type needs a type argument too, i.e. `Any`.

Reply:
Hmm, how come mypy does not complain about this if it's not allowed?

Also, in the already merged PR we do like this multiple times:
```py
    async def async_post_init(
        self, flow: BaseFlowHandler, result: _FlowResultT
    ) -> None:
        """Entry has finished executing its first step asynchronously."""
```

Finally, is this still needed for type variables which have defaults?

Reply:
> Hmm, how come mypy does not complain about this if it's not allowed?

We likely haven't enabled `disallow_any_generics = true` for `data_entry_flow.py` yet. It's an advanced setting. So far only about 30 modules can enable it at all.

`"not allowed"` might be the wrong wording. Missing generics are just interpreted as `Any`. That's why TypeVar defaults are so helpful. E.g. there are a lot of integrations which don't use the generic types for `CoordinatorEntity` and thus instead of `self.coordinator` being inferred as `DataUpdateCoordinator` and `self.coordinator.data` as `dict[str, Any]`, it's just always `Any`. See #111949

> Also, in the already merged PR we do like this multiple times:
> 
> ```python
>     async def async_post_init(
>         self, flow: BaseFlowHandler, result: _FlowResultT
>     ) -> None:
>         """Entry has finished executing its first step asynchronously."""
> ```

Yeah, I mentioned it in one of my earlier comments. Those all should get generic types. https://github.com/home-assistant/core/pull/111952#discussion_r1509254093

> Finally, is this still needed for type variables which have defaults?

No. That's the point of `defaults`. You can leave them out. Just remember that all TypeVars with defaults, must be after TypeVars without them in `Generic[...]`. Otherwise it won't work.